### PR TITLE
Vet Clinic DB | Database Setup #1

### PR DESCRIPTION
**Pull request by PraisesPJMT**
# Vet Clinic Database - Milestone #1 | Database Setup

## Project Description
This project involves the creation of the `vet clinic` database, creating a table to store animals' information, inserting some data into it, and querying it.

## Feature Description
- [Created a database named `vet_clinic`](https://github.com/PraisesPJMT/Veterinary-Clinic-DB/commit/36fd8b6f9efb86d7ea572984a91d49e8523707ce).
- [Created a table named `animals` with the following columns](https://github.com/PraisesPJMT/Veterinary-Clinic-DB/commit/36fd8b6f9efb86d7ea572984a91d49e8523707ce):
  - `id`: integer
  - `name`: string
  - `date_of_birth`: date
  - `escape_attempts`: integer
  - `neutered`: boolean
  - `weight_kg`: decimal
- [Insert the following data into the `animals` table:](https://github.com/PraisesPJMT/Veterinary-Clinic-DB/commit/57881ab9efc0afab833c107300f97abfcadd840c)
  - Animal: His name is Agumon. He was born on Feb 3rd, 2020, and currently weighs 10.23kg. He was neutered and he has never tried to escape.
  - Animal: Her name is Gabumon. She was born on Nov 15th, 2018, and currently weighs 8kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Pikachu. He was born on Jan 7th, 2021, and currently weighs 15.04kg. He was not neutered and he has tried to escape once.
  - Animal: Her name is Devimon. She was born on May 12th, 2017, and currently weighs 11kg. She is neutered and she has tried to escape 5 times.
- [Wrote queries for the following quests:](https://github.com/PraisesPJMT/Veterinary-Clinic-DB/commit/f4421979d57d9aeb7b1c450bfcbea2bb563e3575)
  - Find all animals whose name ends in `mon`.
  - List the name of all animals born between 2016 and 2019.
  - List the name of all neutered animals with less than 3 escape attempts.
  - List the date of birth of all animals named either `Agumon` or `Pikachu`.
  - List name and escape attempts of animals that weigh more than 10.5kg
  - Find all animals that are neutered.
  - Find all animals not named Gabumon.
  - Find all animals with a weight between 10.4kg and 17.3kg (including the animals with weights that equals precisely 10.4kg or 17.3kg)

**Screenshot of the results of my queries can be seen below.**
Screenshot from 2022-10-17 15-03-29

Happy Code Reviewing :+1: :slightly_smiling_face:

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/* Querries on the 'animals' table of 'vet_clinic' */
+
+-- Find all animals whoes name ends in "mon"
+SELECT * FROM animals WHERE name LIKE '%mon';
+
+-- List the name of all animals born between 2016 and 2019
+SELECT name FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-01-01';
```
Comment (ShahierNashaat) [inline queries.sql]: - [x] Kindly notice that you should select the animals that also were born on 2019 so your date should be until `2019-12-31`.

```sql
SELECT name FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-13';
```

File: queries.sql [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+-- List the name of all animals that are neutered and have less than 3 escape attempts
+SELECT name FROM animals WHERE neutered = TRUE AND escape_attempts <= 3;
```
Comment (ShahierNashaat) [inline queries.sql]: - [x] Kindly you should select the animals that are neutered and have escape attempts less than 3 not less than or equal.

```sql
SELECT name FROM animals WHERE neutered = TRUE AND escape_attempts < 3;
```

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @PraisesPJMT, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (PraisesPJMT) [reply queries.sql]: Thanks so much! I did not realise this at first!

File: queries.sql (hunk 2 above)
Comment (PraisesPJMT) [reply queries.sql]: I totally missed that! Thanks so much!

**Comment by PraisesPJMT** [comment]
## Recommended Changes Implemented
(Thanks @ShahierNashaat for the review. Below are the effected recommended feature changes)
- Change the lower limit for the query to `List the name of all animals born between 2016 and 2019` to the last day in 2019
- Change the lower limit for the query to `List the name of all animals that are neutered and have less than 3 escape attempts` to less than 3 escape attempts.

Happy Code Reviewing :+1: :slightly_smiling_face:

**Review by m05e5 — APPROVED** [review]
Hi @PraisesPJMT , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️ You're code is clean. well done 

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**